### DISCUSSION TITLE
How to measure frequency of pulses

### DISCUSSION THREAD

**Discussion by ivanvachovsky** [arendst/Tasmota#21547] [General]
How to measure frequency of pulses coming to a GPIO input.
I have water flow meter that produces pulses proportional to the water flow.
Typical interval between pulses is from 10 to 100 ms. depending on the water flow.
Ideally I'd like Tasmota to report the AVERAGE frequency of the pulses directly to MQTT every 5 seconds (adjustable). 
If the "ideal" is not possible than I would take any reading (like time between pulses for example) but averaging is essential as I don't want to overwhelm the MQTT channel with 100s reports per second.

**Comment by sfromis** [comment]
You can configure a `Counter` input pin, and with `SetOption79` you can have the value reset each TelePeriod, with the count being reported, easily divisible by the interval. However, that's a minimum of 10 seconds.

With ESP32 and the Berry language, you have good features for creating a timed polling of the counter value, and make your calculations.

On ESP8266, you can still use rules to query the counter at timed intervals, pick up the value, find the difference to last time, and divide to get the frequency.

**Comment by ivanvachovsky** [comment]
@sfromis Understood. Will implement and come back if needed. Thanks for the prompt response.
